sway-core/src/semantic_analysis/program.rs

```diff
@@ -196,6 +196,25 @@ impl TypedProgram {
         ok(typed_program_kind, vec![], errors)
     }
 
+    /// Checks that no arguments passed to a `main()` in a `script` or `predicate`.
+    pub(crate) fn check_args_to_main(&self) -> CompileResult<()> {
+        let mut errors = Vec::new();
+        let warnings = Vec::new();
+        match &self.kind {
+            TypedProgramKind::Script { main_function, .. }
+            | TypedProgramKind::Predicate { main_function, .. } => {
+                if !main_function.parameters.is_empty() {
+                    errors.push(CompileError::MainArgsNotYetSupported {
+                        span: main_function.span.clone(),
+                    })
+                }
+            }
+            _ => (),
+        }
+
+        ok((), warnings, errors)
```

Comment:
Looks like event if we detect an error we are still returning `ok(...)`. By looking at the `finalize_types()` we may need to do something like this: 
```suggestion
        if errors.is_empty() {
            ok((), warnings, errors)
        } else {
            err(warnings, errors)
        }
```

Reply:
Oh that makes sense 😅  Thanks a lot !